### ISSUE TITLE
Event overlap

### ISSUE THREAD

**Issue by Sarke** [briannesbitt/Carbon#1016]
Any thoughts on making an Event class, with a start time and duration (or start and end), and then having methods to check if two of them overlap or calculate the time between them?

**Comment by kylekatarnls** [comment]
It would best to be a dedicated library. If you make such library, we would glad to refer it in our documentation.

**Comment by Sarke** [comment]
https://github.com/thephpleague/period

**Comment by kylekatarnls** [comment]
Don't understood you were looking a period class, it exists in Carbon: `CarbonPeriod` (https://carbon.nesbot.com/docs/#api-period). No need to add an other library.

**Comment by Sarke** [comment]
Sorry, I should have been more clear, I meant a Calendar Event, with a start and duration like I mentioned above.

Thanks!

**Comment by Sarke** [comment]
@kylekatarnls Does the Carbon Period have some overlap method?

**Comment by kylekatarnls** [comment]
PHP DateTime (and so Carbon) can be compared with `>`, `<`, etc.

So overlap can be calculated like you would do with numbers:

```php
$range1Start = Carbon::parse('2019-01-26');
$range1End = Carbon::parse('2019-03-03');

$range2Start = Carbon::parse('2019-02-15');
$range2End = Carbon::parse('2019-04-01');

$overlap = $range1End > $range2Start && $range2End > $range1Start;
```

**Comment by kylekatarnls** [comment]
Or with CarbonPeriod:

```php
$range1 = CarbonPeriod::create('2019-01-26', '2019-03-03');
$range2 = CarbonPeriod::create('2019-02-15', '2019-04-01');

$overlap = $range1->getEndDate() > $range2->getStartDate() && $range2->getEndDate() > $range1->getStartDate();
```

If you wish to add a `overlaps` method for that in `CarbonPeriod`, feel free to make a pull-request and I would merge it.

**Comment by kylekatarnls** [comment]
Will be implemented among other CarbonPeriod improvements: #1745
